Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberView.swift

```diff
@@ -8,21 +8,31 @@
 
 import UIKit
 
-protocol CardNumberViewProtocol: AnyObject {
-  var schemeIcon: Constants.Bundle.SchemeIcon { get set }
+enum CardNumberError: Error {
+  case invalid
```

Comment:
I wouldn't create this enum for now to be honest, unless if you know there would be multiple types of error coming in the future. But at this stage, the number will either be valid, or invalid?

So I would remove the property `cardNumberError` from the VC and instead, from within `textFieldShouldEndEditing` you can just straight away call a function
```swift
func showError(_ show: Bool) {
     style.error?.isHidden = !show
}
```
(instead of the current `updateError` from line 73)